src/gpu/intel/ocl/gen9_binary.hpp

```diff
@@ -146,13 +153,22 @@ struct gen9_binary_t : public gpu_primitive_t {
         auto &src1_scale
                 = CTX_IN_STORAGE(DNNL_ARG_SRC_1 | DNNL_ARG_ATTR_SCALES);
 
+        unsigned arg_idx;
         compute::kernel_arg_list_t arg_list;
         arg_list.set(0, src0);
         arg_list.set(1, src1);
-        arg_list.set(2, dst);
 
-        unsigned arg_idx = append_post_ops_to_arg_list(
-                ctx, arg_list, 3, pd()->attr()->post_ops_);
+        if (pd()->is_ternary_op()) {
+            auto &src2 = CTX_IN_STORAGE(DNNL_ARG_SRC_2);
+            arg_list.set(2, src2);
+            arg_list.set(3, dst);
+            arg_idx = append_post_ops_to_arg_list(
+                    ctx, arg_list, 4, pd()->attr()->post_ops_);
+        } else {
+            arg_list.set(2, dst);
+            arg_idx = append_post_ops_to_arg_list(
+                    ctx, arg_list, 3, pd()->attr()->post_ops_);
+        }
```

Comment:
You could do what we do in other places to simplify this logic:
```suggestion
        unsigned arg_idx = 0;
        compute::kernel_arg_list_t arg_list;
        arg_list.set(arg_idx++, src0);
        arg_list.set(arg_idx++, src1);

        if (pd()->is_ternary_op()) {
            auto &src2 = CTX_IN_STORAGE(DNNL_ARG_SRC_2);
            arg_list.set(arg_idx++, src2);
        }

        arg_list.set(arg_idx++, dst);
        arg_idx = append_post_ops_to_arg_list(
                ctx, arg_list, arg_idx, pd()->attr()->post_ops_);
```

`compute::kernel_arg_list_t` also has an `append` method...